casadocodigo/src/main/java/br/com/deveficiente/casadocodigo/livro/NovoLivroForm.java

```diff
@@ -0,0 +1,94 @@
+package br.com.deveficiente.casadocodigo.livro;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import javax.validation.constraints.Future;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import br.com.deveficiente.casadocodigo.categoria.Categoria;
+import br.com.deveficiente.casadocodigo.categoria.CategoriaInexistenteException;
+import br.com.deveficiente.casadocodigo.categoria.CategoriaRepository;
+
+public class NovoLivroForm {
+
+	@NotBlank
+	private String titulo;
+	
+	@NotBlank
+	@Size(max = 500)
+	private String resumo;
+	
+	private String sumario;
+	
+	@NotNull
+	@Min(value = 20)
+	private BigDecimal preco;
+	
+	@NotNull
+	@Min(value = 100)
+	private int numeroPaginas;
+	
+	@NotBlank
+	private String isbn;
+	
+	@Future
+	private LocalDateTime dataPublicacao;
+	
+	@NotNull
+	private Long idCategoria;
+
+	public Livro novoLivro(CategoriaRepository categoriaRepository) {
+		Optional<Categoria> categoria = categoriaRepository.findById(idCategoria);
+		if(!categoria.isPresent()) {
```

Comment:
*Quanto mais externa a borda, mais proteção temos.*

É uma boa aplicação do principio. Uma categoria inexistente tem mais cara de bug do que de comportamento esperado mesmo. Por outro lado, fica como sugestão usar a ```IllegalArgumentException``` para indicar que o argumento é inválido. 

*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Sempre que for possível usar algo pronto, é melhor.

Reply:
Boa, valeu...vou colocar sim.